source/common/quic/client_connection_factory_impl.cc

```diff
@@ -7,12 +7,49 @@
 namespace Envoy {
 namespace Quic {
 
+// A wrapper class for quic::SessionCache.
+class QuicSessionCacheDelegate : public quic::SessionCache {
+public:
+  explicit QuicSessionCacheDelegate(quic::SessionCache& cache) : cache_(cache) {}
+
+  void Insert(const quic::QuicServerId& server_id, bssl::UniquePtr<SSL_SESSION> session,
+              const quic::TransportParameters& params,
+              const quic::ApplicationState* application_state) override {
+    cache_.Insert(server_id, std::move(session), params, application_state);
+  }
+
+  std::unique_ptr<quic::QuicResumptionState>
+  Lookup(const quic::QuicServerId& server_id, quic::QuicWallTime now, const SSL_CTX* ctx) override {
+    auto state = cache_.Lookup(server_id, now, ctx);
+    return state;
+  }
+
+  void ClearEarlyData(const quic::QuicServerId& server_id) override {
+    cache_.ClearEarlyData(server_id);
+  }
+
+  void OnNewTokenReceived(const quic::QuicServerId& server_id, absl::string_view token) override {
+    cache_.OnNewTokenReceived(server_id, token);
+  }
+
+  void RemoveExpiredEntries(quic::QuicWallTime now) override { cache_.RemoveExpiredEntries(now); }
+
+  void Clear() override { cache_.Clear(); }
+
+private:
+  quic::SessionCache& cache_;
```

Comment:
Hm. I think I'm a bit confused about this class. Does it have any different behavior from quic::SessionCache?

Reply:
see above.